### PR TITLE
Feature/rheadley/01.2020.translations

### PR DESCRIPTION
**Pull request by lifewithryan**
# Critical Changes

# Changes
## New Support for Spanish (Mexico)
We've added support for translations in Spanish (Mexico). Now, EDA supports translation in Spanish, Spanish (Mexico), and French, as well as Catalan as a platform-only language.

# Issues Closed

# New Metadata

# Deleted Metadata

# Testing Notes

### CODE REVIEW

File: src/objectTranslations/Account-ca.objectTranslation [hunk 1]
```diff
@@ -7,7 +7,7 @@
         <relationshipLabel>Comptes (adreça actual)</relationshipLabel>
     </fields>
     <fields>
-        <label>Primary Contact</label>
+        <label>_SYSTEM: One2OneContact</label>
```
Comment (lifewithryan) [inline src/objectTranslations/Account-ca.objectTranslation]: note to ryan: we should undo this, it needs to be Primary Contact--my bad.

File: src/objectTranslations/Contact-ca.objectTranslation [hunk 2]
```diff
@@ -3471,66 +3471,38 @@
         <help>Quin número de telèfon s&apos;hauria d&apos;utilitzar per a la majoria de comunicacions que incloguin aquest contacte?</help>
         <label>Telèfon preferit</label>
         <name>PreferredPhone__c</name>
-        <picklistValues>
```
Comment (lifewithryan) [inline src/objectTranslations/Contact-ca.objectTranslation]: Ryan i think we are going to have problems here too.

Comment (lifewithryan) [reply src/objectTranslations/Contact-ca.objectTranslation]: I actually wanna pull @skosgi1 into this bit. Can you verify if this is what it is supposed to be? I feel like this is going to break translation work that we did.

File: src/objectTranslations/Course_Offering__c-en_GB.objectTranslation [hunk 3]
```diff
@@ -48,7 +48,7 @@
     <fields>
         <label>Time Block</label>
         <name>Time_Block__c</name>
-        <relationshipLabel>Time Blocks</relationshipLabel>
```
Comment (lifewithryan) [inline src/objectTranslations/Course_Offering__c-en_GB.objectTranslation]: Ryan this needs to be the  Time Blocks not Course Offerings.

File: src/objectTranslations/Course_Offering__c-es.objectTranslation [hunk 4]
```diff
@@ -48,7 +48,7 @@
     <fields>
         <label>Bloque de tiempo</label>
         <name>Time_Block__c</name>
-        <relationshipLabel>Bloques de tiempo</relationshipLabel>
```
Comment (lifewithryan) [inline src/objectTranslations/Course_Offering__c-es.objectTranslation]: Bloques de Tiempo is the source of truth here.

File: src/objectTranslations/Course_Offering__c-es_MX.objectTranslation [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    </fields>
+    <fields>
+        <label>Plazo</label>
+        <name>Term__c</name>
+        <relationshipLabel>Ofertas de cursos</relationshipLabel>
+    </fields>
+    <fields>
+        <label>Bloque de tiempo</label>
+        <name>Time_Block__c</name>
+        <relationshipLabel>Ofertas de cursos</relationshipLabel>
```
Comment (lifewithryan) [inline src/objectTranslations/Course_Offering__c-es_MX.objectTranslation]: Ryan This is wrong here so perhaps change to placeholder and they'll get it next time.

File: src/objectTranslations/Course_Offering__c-fr.objectTranslation [hunk 6]
```diff
@@ -41,14 +41,14 @@
...
+        <label>Période</label>
         <name>Term__c</name>
-        <relationshipLabel>Offres de cours</relationshipLabel>
+        <relationshipLabel>Choix de modules</relationshipLabel>
     </fields>
     <fields>
         <label>Bloc de temps</label>
         <name>Time_Block__c</name>
-        <relationshipLabel>Blocs de temps</relationshipLabel>
+        <relationshipLabel>Choix de modules</relationshipLabel>
```
Comment (lifewithryan) [inline src/objectTranslations/Course_Offering__c-fr.objectTranslation]: Blocs de temps should be source of truth here.

**Comment by robertbwatson** [comment]
@lifewithryan @juliesnow Do we need to call out anything in the release notes? It looks like we added support for Mexican Spanish?

**Comment by lifewithryan** [comment]
> @lifewithryan @juliesnow Do we need to call out anything in the release notes? It looks like we added support for Mexican Spanish?

good call, that is a worthy call out.

**Comment by lifewithryan** [comment]
> > @lifewithryan @juliesnow Do we need to call out anything in the release notes? It looks like we added support for Mexican Spanish?
> 
> good call, that is a worthy call out.

@juliesnow @robertbwatson I've added a very basic callout for the new translation -- but I'm sure that is not how we want to word it.

File: src/objectTranslations/Behavior_Involvement__c-ca.objectTranslation [hunk 7]
```diff
@@ -9,55 +9,55 @@
         <value><!-- Behavior Involvement --></value>
     </caseValues>
     <fields>
-        <help><!-- The Case that represents the event being reported. --></help>
-        <label><!-- Case --></label>
+        <help>El cas que representa l’esdeveniment denunciat.</help>
```
Comment (robertbwatson) [inline src/objectTranslations/Behavior_Involvement__c-ca.objectTranslation]: @lifewithryan Just out of curiousity, is the ordering here how they are delivering it to us now? If so, that makes such a WORLD of difference!

Comment (lifewithryan) [reply src/objectTranslations/Behavior_Involvement__c-ca.objectTranslation]: I'd have to double check the original branch. I think they might be doing that now but i still had to do an upload and retrieve to fix other items.

Comment (lifewithryan) [reply src/objectTranslations/Behavior_Involvement__c-ca.objectTranslation]: and the answer is no, this was a result of deploying and retrieving.

File: src/objectTranslations/Contact-ca.objectTranslation (hunk 2 above)
Comment (skosgi1) [reply src/objectTranslations/Contact-ca.objectTranslation]: @lifewithryan Sorry noticing this just now...looking into it...FYI @robertbwatson please don't merge this yet.

File: src/objectTranslations/Contact-es.objectTranslation [hunk 8]
```diff
@@ -3471,66 +3471,38 @@
         <help>¿Cuál teléfono debe ser el usado para la mayoría de las comunicaciones con este contacto?</help>
         <label>Teléfono elegido</label>
         <name>PreferredPhone__c</name>
-        <picklistValues>
-            <masterLabel>Home</masterLabel>
-            <translation>Teléfono particular</translation>
-        </picklistValues>
```
Comment (skosgi1) [inline src/objectTranslations/Contact-es.objectTranslation]: @lifewithryan @robertbwatson looks like translations for old picklist values are completely gone.

File: src/objectTranslations/Contact-es.objectTranslation [hunk 9]
```diff
@@ -3471,66 +3471,38 @@
...
         <picklistValues>
             <masterLabel>Home Phone</masterLabel>
-            <translation>Teléfono particular</translation>
+            <translation><!-- Home Phone --></translation>
         </picklistValues>
         <picklistValues>
             <masterLabel>Mobile Phone</masterLabel>
-            <translation>Teléfono móvil</translation>
+            <translation><!-- Mobile Phone --></translation>
         </picklistValues>
```
Comment (skosgi1) [inline src/objectTranslations/Contact-es.objectTranslation]: Also, looks like translations for new picklist values have been removed.

File: unpackaged/post/case_behavior_record_types/package.xml [hunk 10]
```diff
@@ -8,5 +8,13 @@
         	<members>Case.Behavior Management</members>
         	<name>BusinessProcess</name>
 	</types>
+    <types>
```
Comment (robertbwatson) [inline unpackaged/post/case_behavior_record_types/package.xml]: Any reason to not also do this for `unpackaged/post/facility_display_name`?

File: src/package.xml [hunk 11]
```diff
@@ -1097,12 +1126,16 @@
     <types>
         <members>Attribute__c.Credential</members>
         <members>Attribute__c.Student_Characteristic</members>
+        <members>Course_Enrollment__c.Default</members>
```
[the remark this answers is not in the document]
Comment (robertbwatson) [reply src/package.xml]: These got removed when I did a retrieve from the server, so I've addressed this already.

File: unpackaged/post/case_behavior_record_types/package.xml (hunk 10 above)
Comment (lifewithryan) [reply unpackaged/post/case_behavior_record_types/package.xml]: ahhh good catch. I think I removed them from the original pull but forgot to add it to unmanaged. I'll see if we have a translation for it and make sure to put it in.

File: src/objectTranslations/Course_Offering__c-en_US.objectTranslation [hunk 12]
```diff
@@ -48,7 +48,7 @@
     <fields>
         <label><!-- Time Block --></label>
         <name>Time_Block__c</name>
-        <relationshipLabel><!-- Course Offerings --></relationshipLabel>
+        <relationshipLabel><!-- Time Blocks --></relationshipLabel>
```
[the remark this answers is not in the document]
Comment (lifewithryan) [reply src/objectTranslations/Course_Offering__c-en_US.objectTranslation]: my bad, I thought we changed them from Course Offerings to Time Blocks, do I have that info backwards?

File: src/objectTranslations/Course_Offering__c-es_MX.objectTranslation [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+    </fields>
+    <fields>
+        <label>Plazo</label>
+        <name>Term__c</name>
+        <relationshipLabel>Ofertas de cursos</relationshipLabel>
+    </fields>
+    <fields>
+        <label>Bloque de tiempo</label>
+        <name>Time_Block__c</name>
+        <relationshipLabel>Bloques de tiempo</relationshipLabel>
```
[the remark this answers is not in the document]
Comment (lifewithryan) [reply src/objectTranslations/Course_Offering__c-es_MX.objectTranslation]: this is my fault, I thought they changed to time blocks by us but they had old values still reading Course Offerings.

File: src/package.xml [hunk 14]
```diff
@@ -817,21 +817,25 @@
...
         <members>Affiliation__c-en_US</members>
         <members>Affiliation__c-es</members>
+        <members>Affiliation__c-es_MX</members>
         <members>Affiliation__c-fr</members>
         <members>Affl_Mappings__c-ca</members>
         <members>Affl_Mappings__c-en_GB</members>
         <members>Affl_Mappings__c-en_US</members>
         <members>Affl_Mappings__c-es</members>
+        <members>Affl_Mappings__c-es_MX</members>
         <members>Affl_Mappings__c-fr</members>
```
[the remark this answers is not in the document]
Comment (lifewithryan) [reply src/package.xml]: > Do we want placeholders for Application, Test, & Test Score?

@robertbwatson I thought those were getting added in that PR?

File: unpackaged/post/case_behavior_record_types/package.xml (hunk 10 above)
Comment (lifewithryan) [reply unpackaged/post/case_behavior_record_types/package.xml]: I had to add placeholders.

File: src/package.xml (hunk 11 above)
[the remark this answers is not in the document]
Comment (lifewithryan) [reply src/package.xml]: @robertbwatson github is confusing me, is there any action needed here on my part?

File: src/package.xml (hunk 14 above)
[the remark this answers is not in the document]
Comment (lifewithryan) [reply src/package.xml]: could have sworn I just typed an answer here...i thought I saw at least the Test and Test Score translations in your PR for Test and Test score?
In any case, I don't think we'll need them will we? When they do their pull for the next round, they should see they don't currently have translations for them.

File: src/objectTranslations/Course_Offering__c-fr.objectTranslation (hunk 6 above)
Comment (lifewithryan) [reply src/objectTranslations/Course_Offering__c-fr.objectTranslation]: i may be wrong, will verify with team.

File: src/package.xml (hunk 11 above)
[the remark this answers is not in the document]
Comment (robertbwatson) [reply src/package.xml]: No, resolved it on my own with the commit I made last night

File: src/objectTranslations/Course_Offering__c-fr.objectTranslation (hunk 6 above)
Comment (lifewithryan) [reply src/objectTranslations/Course_Offering__c-fr.objectTranslation]: I am wah-rong!

File: src/objectTranslations/Course_Offering__c-en_GB.objectTranslation (hunk 3 above)
Comment (lifewithryan) [reply src/objectTranslations/Course_Offering__c-en_GB.objectTranslation]: I am wrong

**Comment by skosgi1** [comment]
@lifewithryan @robertbwatson fixed translations for preferred email and phone. can you take a quick look when you get a moment? Thx.

File: src/objectTranslations/Course_Offering__c-es_MX.objectTranslation (hunk 13 above)
[the remark this answers is not in the document]
Comment (lifewithryan) [reply src/objectTranslations/Course_Offering__c-es_MX.objectTranslation]: oh, nevermind, yes on th es-MX file, they wouldn't have had this change so they will have to grab this next time.

File: src/objectTranslations/Course_Offering__c-en_US.objectTranslation (hunk 12 above)
[the remark this answers is not in the document]
Comment (lifewithryan) [reply src/objectTranslations/Course_Offering__c-en_US.objectTranslation]: I'm getting myself so confused. Yes they worked on these before we made the fix so they'll have to get those in the next go-round is my guess.

File: src/package.xml (hunk 14 above)
[the remark this answers is not in the document]
Comment (robertbwatson) [reply src/package.xml]: Ah, nevermind - I didn't see them here the first time. They can add the mexican translation files next time, but the others are there now at least.

**Comment by robertbwatson** [comment]
@lifewithryan @skosgi1 I think this looks pretty great now, thanks for your hard work on this.

I only have one more remaining question/comment in the src/objectTranslations/Contact-es_MX.objectTranslation file regarding translation of the Preferred Phone/Preferred Email picklist values for a new language. I wasn't sure if we needed to do anything there re: old values...

Otherwise, I think we're okay to merge. I am a little hesitant though, with how much manual work had to be done, and especially with the Preferred Phone/Preferred Email translations...

**Comment by lifewithryan** [comment]
> @lifewithryan @skosgi1 I think this looks pretty great now, thanks for your hard work on this.
> 
> I only have one more remaining question/comment in the src/objectTranslations/Contact-es_MX.objectTranslation file regarding translation of the Preferred Phone/Preferred Email picklist values for a new language. I wasn't sure if we needed to do anything there re: old values...
> 
> Otherwise, I think we're okay to merge. I am a little hesitant though, with how much manual work had to be done, and especially with the Preferred Phone/Preferred Email translations...

@robertbwatson I understand that feeling all too well. this is our first round of translation merging since fixing that bug. If it break again, we can adjust. I think right now it might not work for es_MX until we get those translations (obviously). I think it is going to be one of those things that just has to lag behind a bit. Now that it's fixed and updated though, I would hope it is not an issue moving forward.

**Comment by lifewithryan** [comment]
@robertbwatson it still says "changes requested" so I want to be sure I've addressed everything. I am ready for you to merge upon your approval. (No pressure or anything...) :)

**Pull request merged**